javascript/powercommand_using_SDKv3/asana_power.js

```diff
@@ -0,0 +1,89 @@
+  .command('incomplete', 'Incomplete all of the matching tasks')
+  .command('comment', 'Add a comment with a message on all of the matching tasks')
+  .alias('t', 'token')
+  .nargs('t', 1)
+  .describe('t', 'The access token used to authenticate to Asana')
+  .alias('p', 'project')
+  .describe('p', 'The project that we should operate on. Can be provided either as an ID or a link')
+  .alias('s', 'save')
+  .describe(
+    's',
+    'Indicates that we should save the provided token and project parameters to the keychain for future requests'
+  )
+  .alias('m', 'message')
+  .describe('m', "The message used in commands such as 'comment'")
+  .alias('a', 'all')
+  .describe(
+    'a',
+    'Indicates that we should operate over all tasks in the target project. Not compatible with other filters'
+  )
+  .alias('n', 'name')
+  .describe(
+    'name',
+    'Indicates that we should operate only on the tasks whose name contains this value. Compatible with other filters'
+  )
+  .alias('o', 'notes')
+  .describe(
+    'notes',
+    'Indicates that we should operate only on the tasks whose notes (the task text) contains this value. Compatible with other filters'
+  )
+  .alias('@', 'assignee')
+  .describe(
+    'assignee',
+    'Indicates that we should operate only on the tasks whose assignee\'s name contains this value. Compatible with other filters'
+  )
+  .alias('c', 'completed')
+  .describe(
+    'completed',
+    'Indicates that we should operate only on the tasks which are completed. Compatible with other filters'
+  )
+  .alias('i', 'incomplete')
+  .describe(
+    'incomplete',
+    'Indicates that we should operate only on the tasks which are incomplete. Compatible with other filters'
+  )
+  .alias('h', 'help')
+  .help('h')
+  .alias('v', 'version').argv;
+const Asana = require('asana');
+const connectionSettings = require('./connection_settings.js');
+const taskProvider = require('./task_provider.js');
+const taskFilterer = require('./task_client_filter.js');
+const commandRunner = require('./command_runner.js');
+
+const run = async () => {
+  try {
+    const projectId = await connectionSettings.getProjectId(argv.project, argv.save);
+    const PAT = await connectionSettings.getToken(argv.token, argv.save);
+    const command = commandRunner.getCommand(argv);
+    if (!command) return;
+    const filters = taskFilterer.getFilters(argv);
+    
+    // Create Asana ApiInstances for tasks and stories using node SDK v3
+    // SDK repo for more information:  https://github.com/Asana/node-asana/
+    const client = Asana.ApiClient.instance;
+    const token = client.authentications['token'];
+    token.accessToken = PAT;
+    const tasksApiInstance = new Asana.TasksApi();
+    const storiesApiInstance = new Asana.StoriesApi();
+
+    const tasks = await taskProvider.getTaskRefs(tasksApiInstance, projectId);
+
+    for(let task of tasks.data) { 
+      if (filters.some(filter => !filter.matchesTask(task))) return;
```

Comment:
This should probably be a `continue` rather than `return` it doesn't make sense to return in this for loop. You want to continue the loop if the task does not match the filter.

The return makes sense in the previous code since it's a  tasks.stream().on(...) takes in a function https://github.com/Asana/devrel-examples/blob/master/javascript/powercommands/asana_power.js#L69-L73

Reply:
If you try to run `asana-power complete --assignee "Kenan Kigunda"` this won't work because the tasks being returned is missing the `assignee.name`. even though you requested for `assignee` in `opt_fields` in your `task_provider.js`https://github.com/Asana/devrel-examples/pull/62/files#diff-c55db41f9d5a798d2282e92481cd6de3642c3ebb4d25c7bb1d48de8ae51315a7R4

The Asana API doesn't not return the full assignee object so it'll error when it tries to filter. Pretty sure this was broken in the v1 as well.

Reply:
Added the extra check. It could be worth coming back in the future if we get meaningful usage to add another request to fetch each task with getTask to get the assignee name. Could also maybe do it with assignee gid.